components/brave_ads/core/internal/common/url/url_util_unittest.cc

```diff
@@ -12,149 +12,234 @@
+  EXPECT_FALSE(ShouldSupportUrl(GURL("https://www.brave.com:1234/thank-you*")));
+  EXPECT_FALSE(ShouldSupportUrl(GURL("https://www.brave.com:1234*")));
+  EXPECT_FALSE(ShouldSupportUrl(GURL("https://brave.com:*/x")));
+  EXPECT_FALSE(ShouldSupportUrl(GURL("https://brave.com:*")));
 }
 
-TEST(BraveAdsUrlUtilTest, DoesNotSupportUrlWithHttpScheme) {
+TEST(BraveAdsUrlUtilTest, ShouldNotSupportIpAddress) {
   // Act & Assert
-  EXPECT_FALSE(DoesSupportUrl(GURL("http://foobar.com")));
+  EXPECT_FALSE(ShouldSupportUrl(GURL("https://1.2.3.4/thank-you*")));
+  EXPECT_FALSE(ShouldSupportUrl(GURL("https://192.168.0.0/*")));
+  EXPECT_FALSE(ShouldSupportUrl(GURL("https://192.*.*.*/x")));
+  EXPECT_FALSE(ShouldSupportUrl(GURL("https://[::1]/thankyou")));
+  EXPECT_FALSE(ShouldSupportUrl(GURL("https://030000001017/")));
+  EXPECT_FALSE(ShouldSupportUrl(GURL("https://0xc000020f/")));
 }
 
-TEST(BraveAdsUrlUtilTest, DoesNotSupportUrlWithFooBarScheme) {
+TEST(BraveAdsUrlUtilTest, ShouldNotSupportUrlWithUsernameOrPassword) {
   // Act & Assert
-  EXPECT_FALSE(DoesSupportUrl(GURL("foobar://baz")));
+  EXPECT_FALSE(ShouldSupportUrl(GURL("https://user:pwd@brave.com/thank-you*")));
+  EXPECT_FALSE(ShouldSupportUrl(GURL("https://user@brave.com:1234*")));
+  EXPECT_FALSE(ShouldSupportUrl(GURL("https://*@brave.com/x")));
+  EXPECT_FALSE(ShouldSupportUrl(GURL("https://user:*@brave.com/y")));
+}
+
+TEST(BraveAdsUrlUtilTest, ShouldNotSupportUrlWithWildcardInEtldPlus1) {
+  // Act & Assert
+  EXPECT_FALSE(ShouldSupportUrl(GURL("https://www.google.co.*")));
+  EXPECT_FALSE(ShouldSupportUrl(GURL("https://www.google.*.uk")));
+  EXPECT_FALSE(ShouldSupportUrl(GURL("https://www.*.co.uk")));
+  EXPECT_FALSE(ShouldSupportUrl(GURL("https://www.google.co.uk*")));
+  EXPECT_FALSE(
+      ShouldSupportUrl(GURL("https://www.comparecredit.com*/secure*")));
+  EXPECT_FALSE(ShouldSupportUrl(GURL("https://www.sophos.com*thank-you*")));
+  EXPECT_FALSE(ShouldSupportUrl(GURL("https://*.security/*")));
+  EXPECT_FALSE(ShouldSupportUrl(GURL("https://*.appspot.com/*")));
+}
+
+TEST(BraveAdsUrlUtilTest, ShouldNotSupportLocalUrl) {
+  // Act & Assert
+  EXPECT_FALSE(ShouldSupportUrl(GURL("https://localhost/*")));
+  EXPECT_FALSE(ShouldSupportUrl(GURL("https://example.local/*")));
+  EXPECT_FALSE(ShouldSupportUrl(GURL("https://*.example.local/*")));
+  EXPECT_FALSE(ShouldSupportUrl(GURL("https://*.local/*")));
+  EXPECT_FALSE(ShouldSupportUrl(GURL("https://localhost*/")));
+}
+
+TEST(BraveAdsUrlUtilTest, ShouldSupportUrl) {
+  // Act & Assert
+  EXPECT_TRUE(ShouldSupportUrl(GURL("https://*.google.co.uk")));
+  EXPECT_TRUE(ShouldSupportUrl(GURL("https://www.google.co.uk/*")));
+  EXPECT_TRUE(
+      ShouldSupportUrl(GURL("https://dashboard.0x.org/create-account/"
+                            "verification-sent?tx-relay-brave*")));
```

Comment:
@tmancey if these URLs are getting passed through chromium's match pattern logic, do `?`'s need to be escaped? https://github.com/brave/chromium/blob/main/base/strings/pattern.h

otherwise `https://dashboard.0x.org/create-account/verification-sent?tx-relay-brave*` would match `https://dashboard.0x.org/create-account/verification-senttx-relay-brave*` which is not what the advertiser wants.

Reply:
Thanks. Added